olive/common/ort_inference.py

```diff
@@ -52,7 +52,9 @@ def get_ort_inference_session(
         if ep == "QNNExecutionProvider":
             # add backend_path for QNNExecutionProvider
             execution_provider[idx] = ("QNNExecutionProvider", {"backend_path": "QnnHtp.dll"})
-            break
+        if isinstance(ep, list) and len(ep) == 2 and isinstance(ep[1], dict):
```

Comment:
>if isinstance(ep, list) and len(ep) == 2 and isinstance(ep[1], dict) [](http://example.com/codeflow?start=8&length=68)

what's the purpose of this line? are there any examples or which part could trigger it?

Reply:
When we load a cached model that has a tuned inference setting, the tuple is loaded as a list like `["CPUExecutionProvider", {}]`. The ort runtime session generates a warning saying the format is not right. So, I added this line to correctly parse it to a tuple.

Reply:
You can see the tuple being created by the perf tuner here https://github.com/microsoft/Olive/blob/main/olive/passes/onnx/perf_tuning.py#L110

Reply:
I can also remove this change from this PR since it is unrelated to it. We can revisit this issue later. I also don't think we should be ignoring the second element in tuples here https://github.com/microsoft/Olive/blob/main/olive/common/ort_inference.py#L47.

Reply:
Removed it and created a work item to track it.